### PR TITLE
Added note about ASPNETCORE projects

### LINKED ISSUE #13455: Undocumented required NuGet reference
According to [this response from your team member](https://github.com/dotnet/sdk/issues/2253#issuecomment-512834832), all unit test project for AspNetCore needs to explicitly reference `Microsoft.AspNetCore.App` to avoid reference conflicts.
This is clearly indicated as a requirement in the [integration testing documentation](https://docs.microsoft.com/en-us/aspnet/core/test/integration-tests?view=aspnetcore-2.2#test-app-prerequisites) but not over there... Perhaps not a big deal bug this took me a couple hours to sort things out...

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: e39c2f1b-f41a-bdf9-2646-09379941453d
* Version Independent ID: 422811cb-6287-03af-ba8d-1daebee9744d
* Content: [Unit testing C# code in .NET Core using dotnet test and xUnit - .NET Core](https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-with-dotnet-test?toc=%2Faspnet%2Fcore%2Ftoc.json&bc=%2Faspnet%2Fcore%2Fbreadcrumb%2Ftoc.json&view=aspnetcore-2.2#creating-the-test-project)
* Content Source: [docs/core/testing/unit-testing-with-dotnet-test.md](https://github.com/dotnet/docs/blob/master/docs/core/testing/unit-testing-with-dotnet-test.md)
* Product: **dotnet-core**
* GitHub Login: @ardalis
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by Thraka**
## Summary

Add note about using unit tests on ASPNETCORE apps.

Fixes #13455

### CODE REVIEW

**Comment by Thraka** [comment]
@mairaw Can you do a quick scan of this? Thanks 😄

**Review by mairaw — CHANGES_REQUESTED** [review]
Looks good. If you're using the same text over and over, you should use an include file instead of repeating the same text in each file.

File: docs/core/testing/unit-testing-fsharp-with-mstest.md [hunk 1]
```diff
@@ -64,6 +64,7 @@ The test project requires other packages to create and run unit tests. `dotnet n
 dotnet add reference ../MathService/MathService.fsproj
 ```
 
+[!INCLUDE [testing an ASP.NET Core project from .NET Core](~/includes/selective-unit-tests.md)]
```
Comment (Thraka) [inline docs/core/testing/unit-testing-fsharp-with-mstest.md]:
```suggestion
[!INCLUDE [testing an ASP.NET Core project from .NET Core](~/includes/selective-unit-tests.md)]

```

File: docs/core/testing/unit-testing-fsharp-with-dotnet-test.md [hunk 2]
```diff
@@ -64,6 +64,8 @@ The test project requires other packages to create and run unit tests. `dotnet n
 dotnet add reference ../MathService/MathService.fsproj
 ```
 
+[!INCLUDE [testing an ASP.NET Core project from .NET Core](~/includes/selective-unit-tests.md)]
```
Comment (mairaw) [inline docs/core/testing/unit-testing-fsharp-with-dotnet-test.md]: Same comment applies to all files.
Although ~ is more practical, when I can, I use the relative ones to make it easier for people to find the files when coming to edit them on GitHub from docs. And the new file is called core-testing-note-aspnet.md, right?

```suggestion
[!INCLUDE [testing an ASP.NET Core project from .NET Core](../../../includes/core-testing-note-aspnet.md)]
```

**Review by mairaw — CHANGES_REQUESTED** [review]

File: docs/core/testing/unit-testing-fsharp-with-dotnet-test.md (hunk 2 above)
Comment (Thraka) [reply docs/core/testing/unit-testing-fsharp-with-dotnet-test.md]: Weird, i dont know why that name got in there.

Comment (Thraka) [reply docs/core/testing/unit-testing-fsharp-with-dotnet-test.md]: strange, it's fine on my machine. there must have been a desync with github. I'll reset my local branch.

**Comment by mairaw** [comment]
The changes LGTM but reading the issue more carefully now, I'm not sure if we should add this to these articles @Thraka. These articles don't show web projects at all. @Rick-Anderson do you have any articles that cover unit testing ASP.NET Core projects?

While I'm reading this for example (https://review.docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-fsharp-with-dotnet-test?branch=pr-en-us-13490), it doesn't make any sense to add those packages where you're mentioning, unless I'm completely misunderstanding. @nguerrera can you please take a look?
Perhaps the place where you could mention this requirement is here:
https://docs.microsoft.com/en-us/dotnet/core/testing/

**Comment by Rick-Anderson** [comment]
> do you have any articles that cover unit testing ASP.NET Core projects?

[Unit tests of controller logic](https://docs.microsoft.com/en-us/aspnet/core/mvc/controllers/testing?view=aspnetcore-2.2#unit-tests-of-controller-logic)

**Comment by Thraka** [comment]
This is the main node for Unit Testing in the ASP.NET Core repo, which I believe is why the customer opened the issue: https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-with-dotnet-test?toc=%2Faspnet%2Fcore%2Ftoc.json&bc=%2Faspnet%2Fcore%2Fbreadcrumb%2Ftoc.json&view=aspnetcore-2.2

It's a .NET Core article but it's listed in the ASP.NET repo.

File: includes/core-testing-note-aspnet.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+>[!IMPORTANT]
+>If you're _not_ using the **.NET Core 3.0 SDK**, you must add two additional package references, `Microsoft.AspNetCore.App` and `Microsoft.AspNetCore.Mvc.Testing`.
```
Comment (nguerrera) [inline includes/core-testing-note-aspnet.md]: You will need to add the Microsoft.AspNetCore.App if TargetFramework is less than netcoreapp3.0, even if you are building with the 3.0 SDK.

File: includes/core-testing-note-aspnet.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+>
+>```
+>dotnet add package Microsoft.AspNetCore.App
```
Comment (nguerrera) [inline includes/core-testing-note-aspnet.md]: This should be added without a package version. Does `dotnet add package` respect that? I imagine you'd get a version and then get a warning in the build.

File: includes/core-testing-note-aspnet.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+>dotnet add package Microsoft.AspNetCore.Mvc.Testing
```
Comment (nguerrera) [inline includes/core-testing-note-aspnet.md]: This is the first time I've seen this package as a requirement here. Is it always required?

**Comment by nguerrera** [comment]
> While I'm reading this for example (https://review.docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-fsharp-with-dotnet-test?branch=pr-en-us-13490), it doesn't make any sense to add those packages where you're mentioning, unless I'm completely misunderstanding. @nguerrera can you please take a look?

I agree with this. That page looks very confusing now.

File: includes/core-testing-note-aspnet.md (hunk 4 above)
Comment (Thraka) [reply includes/core-testing-note-aspnet.md]: It adds whatever version is the latest that it uses. I can't see an easy way to force it to versionless apart from editing the csproj file directly.

File: includes/core-testing-note-aspnet.md (hunk 5 above)
Comment (Thraka) [reply includes/core-testing-note-aspnet.md]: According to https://docs.microsoft.com/en-us/aspnet/core/test/integration-tests?view=aspnetcore-2.2#test-app-prerequisites it's required.

**Comment by Thraka** [comment]
> > While I'm reading this for example (https://review.docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-fsharp-with-dotnet-test?branch=pr-en-us-13490), it doesn't make any sense to add those packages where you're mentioning, unless I'm completely misunderstanding. @nguerrera can you please take a look?
> 
> I agree with this. That page looks very confusing now.

Should we just opt out of these details here and move it to the overview?

File: includes/core-testing-note-aspnet.md (hunk 4 above)
Comment (nguerrera) [reply includes/core-testing-note-aspnet.md]: cc @dsplaisted. We should be careful recommending this, then. 

I forget what happens when you have a version, does it win or just warn?

Comment (dsplaisted) [reply includes/core-testing-note-aspnet.md]: It should warn

Comment (nguerrera) [reply includes/core-testing-note-aspnet.md]: warn and then ignore or warn and then use what you say?

File: includes/core-testing-note-aspnet.md (hunk 5 above)
Comment (Thraka) [reply includes/core-testing-note-aspnet.md]: After reading a bit more about this, I believe you're correct. This package is only needed to supplement MVC testing which isn't pertinent to this note.

**Comment by mairaw** [comment]
We don't always have to implement exactly what an issue is requesting. I think if ASP.NET Core has a home for unit testing, you could just a note about specific requirements for certain project types and link to it and then they'll figure out their own requirements.

**Comment by Thraka** [comment]
I think it would be more appropriate for the ASP.NET docs to remove this article from the TOC and create one specifically geared towards unit testing ASP.NET.

@scottaddie what do you think?

File: includes/core-testing-note-aspnet.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
+>[!IMPORTANT]
+>If you're testing an ASP.NET application or library, and you are targeting a framework _other_ than `netcoreapp3.0`, you must add an additional package reference: `Microsoft.AspNetCore.App`.
```
Comment (scottaddie) [inline includes/core-testing-note-aspnet.md]: I think this sentence needs some clarification. `netcoreapp3.0` is the ONLY valid TFM as of ASP.NET Core 3.0. ASP.NET Core 2.2 and earlier versions allow you to target either .NET Framework or .NET Core.

```suggestion
>If you're testing an ASP.NET Core application, and you're targeting a framework _other_ than `netcoreapp3.0`, you must add an additional package reference: `Microsoft.AspNetCore.App`.
```

**Comment by scottaddie** [comment]
@Thraka I'd personally go with Maira's suggestion above.

File: includes/core-testing-note-aspnet.md [hunk 7]
```diff
@@ -0,0 +1 @@
+This article is about testing a .NET Core project. If you're testing an **ASP.NET Core** project, see [Integration tests in ASP.NET Core](/aspnet/core/test/integration-tests?view=aspnetcore-2.2#test-app-prerequisites).
```
Comment (scottaddie) [inline includes/core-testing-note-aspnet.md]:
```suggestion
This article is about testing a .NET Core project. If you're testing an **ASP.NET Core** project, see [Integration tests in ASP.NET Core](/aspnet/core/test/integration-tests#test-app-prerequisites).
```

File: 3.0planning.txt [hunk 8]
```diff
@@ -0,0 +1,100 @@
+.NET Core 3.0 can be used to build great applications for Windows Desktop
```
Comment (mairaw) [inline 3.0planning.txt]: Why these files are in this PR @Thraka?

**Comment by mairaw** [comment]
I still don't feel the PR is quite there (ignoring the other issues that you have with unrelated files).

- I think you need to add a note about different requirements and link to the ASP.NET Core from https://review.docs.microsoft.com/en-us/dotnet/core/testing/?branch=pr-en-us-13490
- I think that the different tutorials could have their title changed to more clearly indicate that this is a tutorial for a .NET Core app. Then, you wouldn't even need the include. On the see also section then, you could add links to the ASP.NET Core tutorial.

This also reminds me that we had it in our backlog to combine all these tutorials into one for each testing framework. I'll add a vendor task for that.

**Comment by Thraka** [comment]
> I still don't feel the PR is quite there (ignoring the other issues that you have with unrelated files).
> 
> * I think you need to add a note about different requirements and link to the ASP.NET Core from https://review.docs.microsoft.com/en-us/dotnet/core/testing/?branch=pr-en-us-13490
> * I think that the different tutorials could have their title changed to more clearly indicate that this is a tutorial for a .NET Core app. Then, you wouldn't even need the include. On the see also section then, you could add links to the ASP.NET Core tutorial.
> 
> This also reminds me that we had it in our backlog to combine all these tutorials into one for each testing framework. I'll add a vendor task for that.

Looks like github failed me and pulled in some files. I'll get those removed. I know we had an issue about redoing these test articles.

**Comment by Thraka** [comment]
@mairaw OK I got all the dumb git conflicts and issues fixed. Do you have an idea on changing the titles? Should we just merge this and then wait for the other issue that changes all of the articles to consolidate?

**Review by mairaw — APPROVED** [review]
LGTM

**Comment by mairaw** [comment]
And yes, we can think of title changes later. I like the changes now! Thanks.

**Pull request merged**